src/service/fixed-layer.js

```diff
@@ -157,16 +173,10 @@ export class FixedLayer {
     }
 
     // Some of the elements may no longer be in DOM.
-    /** @type {!Array<number>} */
-    const toRemove = [];
-    this.fixedElements_.forEach((fe, i) => {
-      if (!this.doc.contains(fe.element)) {
-        toRemove.push(i);
-      }
-    });
-    toRemove.forEach(i => {
-      this.fixedElements_.splice(i, 1);
-    });
+    /** @type {!Array<!FixedElementDef>} */
+    const toRemove = this.fixedElements_.filter(
+        fe => !this.doc.contains(fe.element));
+    toRemove.forEach(fe => this.removeFixedElement_(fe.element));
```

Comment:
nice, i would also have done:

```js
this.fixedElements.splice(0, 9e9, ...(this.fixedElements_.filter(fe => this.doc.contains(fe.element)));
```

no idea if this is any faster (1 large splice vs n splices), but just showing an alternative. probably doesn't matter since the the set is probably small anyways.

im sure @jridgewell would know